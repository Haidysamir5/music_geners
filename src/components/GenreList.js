import React, { Component } from "react";
import { connect } from "react-redux";
import { fetchGenre } from "../actions";
import GenreItem from "./GenreITem";

class GenreList extends Component {

  componentDidMount() {
    this.props.fetchGenre(); //fetch genre on mount
  }

  renderGenre() {
    var { genres } = this.props;
    //loop on genres and get genreitem of each one
    if(genres.length > 0){
      return genres.map((genre) => {
        return <GenreItem props={genre}  key={genre.id}/>; 
      });
    }
    return "...loading";
  }
  render() {
    return (
      <div className="container">
        <div className="row">
          <div className="col-12">
            <div  className="mt-2 mb-2" style={{height:"50px ", background:"#dc2878"}}>
              <h5 className="p-2 text-white"> Genre Types</h5>
            </div>
          </div>
        </div>
        <div className="row">{this.renderGenre()}</div>
      </div>
    );
  }
}

function mapStateToProps(state) {
  return { genres: state.genres };
}

export default connect(mapStateToProps, { fetchGenre })(GenreList);

