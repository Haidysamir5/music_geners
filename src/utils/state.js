export const music={
    modal_state:false,
    music_info:[]
}