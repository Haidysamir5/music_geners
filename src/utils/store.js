import { createStore, applyMiddleware } from "redux";
import promiseMiddleware from "redux-promise";
import reducers from "../reducers";
export const store = createStore(reducers, applyMiddleware(promiseMiddleware));

