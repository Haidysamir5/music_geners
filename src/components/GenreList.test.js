import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import React from "react";
import { act } from "react-dom/test-utils";
import GenreList from "./GenreList";
// wrap the root component in a Provider
import { store } from "../utils/store";

const fetchGenre = jest.fn();
jest.mock("../actions", async () => {
  return {
    __esModule: true,
    default: async () =>  { props:[
      
      {
        id: 0,
        name: "All",
        picture_big: "https.jpg",
        type: "genre",
      },
      {
        id: 132,
        name: "Pop",
        picture_big: "https:.jpg",

        type: "genre",
      }
    ]}
  };
});

test("match snapshoot", async () => {
  let wrapper;
  await act(async () => {
    wrapper = mount(<GenreList store={store} fetchGenre={fetchGenre}/>);
  });
  wrapper.update();

  expect(toJson(wrapper)).toMatchSnapshot();
});
