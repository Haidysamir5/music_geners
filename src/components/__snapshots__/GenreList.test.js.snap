// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshoot 1`] = `
<Connect(GenreList)
  fetchGenre={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <GenreList
    fetchGenre={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    genres={Array []}
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-12"
        >
          <div
            className="mt-2 mb-2"
            style={
              Object {
                "background": "#dc2878",
                "height": "50px ",
              }
            }
          >
            <h5
              className="p-2 text-white"
            >
               Genre Types
            </h5>
          </div>
        </div>
      </div>
      <div
        className="row"
      >
        ...loading
      </div>
    </div>
  </GenreList>
</Connect(GenreList)>
`;
